### PR TITLE
Changelog for v1.15.0-rc1

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,230 @@ This changelog goes through all the changes that have been made in each release
...
+  * Creating a repo from a template repo via API (#15958)
+  * Add Active and ProhibitLogin to API (#15689)
+  * Add Location, Website and Description to API (#15675)
+  * Expose resolver via API (#15167)
+  * [refactor] Unify the export of user data via API (#15144)
+* BUGFIXES
+  * Fix external renderer (#16401)
+  * Fix invalid params and typo of email templates (#16394)
+  * Fix #16387 - rootless Docker user (#16388)
+  * fix activation of primary email addresses (#16385)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Fix activation of primary email addresses (#16385)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,230 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix calculation for finalPage in repo-search component (#16382)
+  * specify user in rootless container numerically (#16361)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Specify user in rootless container numerically (#16361)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,230 @@ This changelog goes through all the changes that have been made in each release
...
+  * Escape reference to `user` table in models.SearchEmails (#16313)
+  * Fix default push instructions on empty repos (#16302)
+  * Fix modified files list in webhooks when there is a space (#16288)
+  * Fix webhook commits wrong hash on HEAD reset (#16283)
+  * Fuzzer finds an NPE due to incorrect URLPrefix (#16249)
+  * Don't WARN log UserNotExist errors on ExternalUserLogin failure (#16238)
+  * Do not show No match found for tribute (#16231)
+  * Fix "Copy Link" for pull requests (#16230)
+  * Fix diff expansion is missing final line in a file (#16222)
+  * v180 migration should be standalone (#16151)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,230 @@ This changelog goes through all the changes that have been made in each release
...
+  * [refactor] mailer service (#15072)
+  * Counterwork seemingly unclickable repo button labels (#15064)
+  * Add DefaultMergeStyle option to repository (#14789)
+  * Added support for gopher URLs. (#14749)
+  * Rework repository archive (#14723)
+  * Add links to toggle WIP status (#14677)
+  * Add Tabular Diff for CSV files (#14661)
+  * Use milestone deadline when sorting issues (#14551)
+* TESTING
+  * just add some unit tests (#16291)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add tag protection (#15629)
+  * Add push to remote mirror repository (#15157)
+  * Add Image Diff for SVG files (#14867)
+  * Add dashboard milestone search and repo milestone search by name. (#14866)
+  * Add LFS Migration and Mirror (#14726)
+  * Improve notifications for WIP draft PR's (#14663)
+  * Disable Stars config option (#14653)
+  * Add option to provide signature for a token to verify key ownership (#14054)
+  * OAuth2 auto-register (#5123)
+* SECURITY
```
Comment (techknowlogick) [inline CHANGELOG.md]: Can you move `SECURITY` above `FEATURES`?

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * GET / SET User Settings (#16169)
+  * expose repo.GetReviewers() & repo.GetAssignees() (#16168)
+  * User expose counters (#16167)
+  * Add repoGetTag (#16166)
+  * Add repoCreateTag (#16165)
+  * Creating a repo from a template repo via API (#15958)
+  * Add Active and ProhibitLogin to API (#15689)
+  * Add Location, Website and Description to API (#15675)
+  * Expose resolver via API (#15167)
+  * [refactor] Unify the export of user data via API (#15144)
```
Comment (techknowlogick) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Comment out app.example.ini (#15807)
+  * Improve logo customization docs (#15754)
+  * Add some reponse status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Update Vendor (#16325)
+  * Speed up git diff highlight generation (#16180)
+  * Remove utf8 option from installation page (#16126)
+  * Vendor Update (#16121)
```
Comment (techknowlogick) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Stop packaging node_modules in release tarballs (#15273)
+  * Introduce esbuild on webpack (#14578)
+* DOCS
+  * Comment out app.example.ini (#15807)
+  * Improve logo customization docs (#15754)
+  * Add some reponse status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Update Vendor (#16325)
```
Comment (techknowlogick) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Improve performance of dashboard list orgs (#16099)
+  * Use git log name-status in get last commit (#16059)
+  * Fix unknown translation key (#16057)
+  * Use Wants= over Requires= in systemd file (#15897)
+  * Update JS dependencies, use Node 16 (#15804)
+  * Add trace logging to SSO methods (#15803)
+  * Hold the event source when there are no listeners (#15725)
+  * Multiple Queue improvements: LevelDB Wait on empty, shutdown empty shadow level queue, reduce goroutines etc (#15693)
+  * Add NeedPostProcess for Parser interface to improve performance of csv parser and some external parser (#15153)
+  * Update JS dependencies (#15033)
```
Comment (techknowlogick) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Update Vendor (#16325)
+  * Speed up git diff highlight generation (#16180)
+  * Remove utf8 option from installation page (#16126)
+  * Vendor Update (#16121)
+  * Improve performance of dashboard list orgs (#16099)
+  * Use git log name-status in get last commit (#16059)
+  * Fix unknown translation key (#16057)
```
Comment (techknowlogick) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Improve logo customization docs (#15754)
+  * Add some reponse status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Update Vendor (#16325)
+  * Speed up git diff highlight generation (#16180)
+  * Remove utf8 option from installation page (#16126)
+  * Vendor Update (#16121)
+  * Improve performance of dashboard list orgs (#16099)
```
Comment (techknowlogick) [inline CHANGELOG.md]: enhancement?

File: CHANGELOG.md [hunk 12]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Use git log name-status in get last commit (#16059)
```
Comment (techknowlogick) [inline CHANGELOG.md]: enhancement?

File: CHANGELOG.md [hunk 13]
```diff
@@ -4,6 +4,228 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix unknown translation key (#16057)
+  * Use Wants= over Requires= in systemd file (#15897)
+  * Update JS dependencies, use Node 16 (#15804)
```
Comment (techknowlogick) [inline CHANGELOG.md]: Move to build, and only mention the use Node 16

**Review by techknowlogick — COMMENTED** [review]
a few nits, also each line should start with a capital letter unless it otherwise makes sense.

File: CHANGELOG.md [hunk 14]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix Benchmark tests, remove a broken one & add two new  (#15250)
+  * Create Proper Migration tests (#15116)
+* TRANSLATION
+  * Use a special name for update default branch on repository setting (#15893)
+  * Fix mirror_lfs source string in en-US locale (#15369)
+* BUILD
+  * Upgrade xorm to v1.1.1 (#16339)
+  * Alpine 3.14 released (#16170)
+  * Disable legal comments in esbuild (#15929)
+  * Use esbuild to minify CSS (#15756)
```
Comment (6543) [inline CHANGELOG.md:203]:
```suggestion
  * Switch to Node 16 to build fronted  (#15804)
  * Use esbuild to minify CSS (#15756)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Comment out app.example.ini (#15807)
+  * Improve logo customization docs (#15754)
+  * Add some response status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Speed up git diff highlight generation (#16180)
+  * Remove utf8 option from installation page (#16126)
+  * Use Wants= over Requires= in systemd file (#15897)
+  * Update JS dependencies, use Node 16 (#15804)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Use git log name-status in get last commit (#16059)
+  * Fix 500 Error with branch and tag sharing the same name (#16040)
+  * Fix get tag when migration (#16014)
+  * Add custom emoji support (#16004)
+  * Kd/fix release compare extra bracket (#15988)
+  * Use filepath.ToSlash and Join in indexer defaults and queues (#15971)
+  * Add permission check for ``GenerateRepository`` (#15946)
+  * Ensure settings for Service and Mailer are read on the install page (#15943)
+  * Fix layout of milestone view (#15927)
+  * Unregister non-matching serviceworkers (#15834)
```
[the remark this answers is not in the document]
Comment (lafriks) [reply CHANGELOG.md:177]: no need to include changes that was already backported

File: CHANGELOG.md [hunk 17]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Change default queue settings to be low go-routines (#15964)
+  * Improve assets handler middleware (#15961)
+  * Rename StaticUrlPrefix to AssetUrlPrefix (#15779)
+  * Use a generic markup class to display externally rendered files and diffs (#15735)
+  * Add frontend testing, require node 12 (#15315)
+  * Add `/assets` as root dir of public files (#15219)
+  * use level config in log section when sub log section not set level (#15176)
+  * Links in markdown should be absolute to the repository not the server (#15088)
+* SECURITY
+  * Support custom mime type mapping for text files (#16304)
```
Comment (jpraet) [inline CHANGELOG.md]: Is this really a security change? I think it's a feature, that has security implications to consider when enabling it. But it's not enabled by default.

File: CHANGELOG.md [hunk 18]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add some response status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Speed up git diff highlight generation (#16180)
+  * Remove utf8 option from installation page (#16126)
+  * Use Wants= over Requires= in systemd file (#15897)
+  * Add trace logging to SSO methods (#15803)
+  * Hold the event source when there are no listeners (#15725)
+  * Multiple Queue improvements: LevelDB Wait on empty, shutdown empty shadow level queue, reduce goroutines etc (#15693)
```
Comment (zeripath) [inline CHANGELOG.md]: Enhancement I think.

File: CHANGELOG.md [hunk 19]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Comment out app.example.ini (#15807)
+  * Improve logo customization docs (#15754)
+  * Add some response status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Speed up git diff highlight generation (#16180)
+  * Remove utf8 option from installation page (#16126)
+  * Use Wants= over Requires= in systemd file (#15897)
+  * Add trace logging to SSO methods (#15803)
```
Comment (zeripath) [inline CHANGELOG.md]: Enhancement

File: CHANGELOG.md [hunk 20]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Stop packaging node_modules in release tarballs (#15273)
+  * Introduce esbuild on webpack (#14578)
+* DOCS
+  * Comment out app.example.ini (#15807)
+  * Improve logo customization docs (#15754)
+  * Add some response status on api docs (#15399)
+  * Rework Token API comments (#15162)
+  * Add better errors for disabled account recovery (#15117)
+* MISC
+  * Speed up git diff highlight generation (#16180)
```
Comment (zeripath) [inline CHANGELOG.md]: Enhancement

File: CHANGELOG.md [hunk 21]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Remove utf8 option from installation page (#16126)
+  * Use Wants= over Requires= in systemd file (#15897)
+  * Add trace logging to SSO methods (#15803)
+  * Hold the event source when there are no listeners (#15725)
```
Comment (zeripath) [inline CHANGELOG.md]: Enhancement

File: CHANGELOG.md [hunk 22]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Multiple Queue improvements: LevelDB Wait on empty, shutdown empty shadow level queue, reduce goroutines etc (#15693)
+  * Add NeedPostProcess for Parser interface to improve performance of csv parser and some external parser (#15153)
```
Comment (zeripath) [inline CHANGELOG.md]: Enhancement

File: CHANGELOG.md [hunk 23]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add code block highlight to orgmode back (#14222)
```
Comment (zeripath) [inline CHANGELOG.md]: Bug fix

File: CHANGELOG.md [hunk 24]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * More efficiently parse shas for shaPostProcessor (#16101)
+  * Fix `doctor --run check-db-consistency --fix` with label fix (#16094)
+  * Prevent webhook action buttons from shifting (#16087)
+  * Change default TMPDIR path in rootless containers (#16077)
+  * Fix typo and add TODO notice (#16064)
+  * Use git log name-status in get last commit (#16059)
+  * Fix 500 Error with branch and tag sharing the same name (#16040)
+  * Fix get tag when migration (#16014)
+  * Add custom emoji support (#16004)
+  * Kd/fix release compare extra bracket (#15988)
```
[the remark this answers is not in the document]
Comment (zeripath) [reply CHANGELOG.md]: nope it should be skipped - I've marked the PR skip-changelog.

File: CHANGELOG.md [hunk 25]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Upgrade xorm to v1.1.1 (#16339)
+  * Alpine 3.14 released (#16170)
+  * Disable legal comments in esbuild (#15929)
+  * Switch to Node 16 to build fronted  (#15804)
+  * Use esbuild to minify CSS (#15756)
+  * Use binary version of revive linter (#15739)
+  * Fix: npx webpack make: *** [Makefile:699: public/js/index.js] Error -… (#15465)
+  * Stop packaging node_modules in release tarballs (#15273)
+  * Introduce esbuild on webpack (#14578)
+* DOCS
```
[the remark this answers is not in the document]
Comment (zeripath) [reply CHANGELOG.md:208]: It's just providing the docs for another PR that is already in the changelog so I'm not certain how much the changelog is improved by its inclusion

File: CHANGELOG.md (hunk 17 above)
Comment (zeripath) [reply CHANGELOG.md]: Agreed it's an enhancement. I've changed the label on the PR.

File: CHANGELOG.md (hunk 24 above)
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

**Review by techknowlogick — APPROVED** [review]

File: CHANGELOG.md [hunk 26]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.io).
 
+## [1.15.0-rc1](https://github.com/go-gitea/gitea/releases/tag/v1.15.0-rc1) - 2021-07-14
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
## [1.15.0-rc1](https://github.com/go-gitea/gitea/releases/tag/v1.15.0-rc1) - 2021-07-15
```

**Comment by fnetX** [comment]
Just a side note: What about including a quick Heads Up section what people would need to "adjust" in order to install / maintain / use the software. IMHO the "BREAKING" section is rather targeted on people who follow the development of Gitea and acts as a reminder what has been changed in this release / RC.
But for normal users, I suppose this isn't really an understandable thing. Just one example, Codeberg might not have been the only installation to place the robots.txt into the public folder to get it served (instead of the CustomPath as told in the docs), and some admin might not catch to [move the robots.txt file](https://codeberg.org/Codeberg/Community/issues/430#issuecomment-240814). I know this is implicitly mentioned when serving assets from another folder, but still, I think reading your changelog doesn't always make it obvious how the change will affect you (as user, admin, ...).

Thank you for considering this.

File: CHANGELOG.md [hunk 27]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Make app.ini permissions more restrictive (#16266)
+  * Refactor Webhook + Add X-Hub-Signature (#16176)
+  * Add asymmetric JWT signing (#16010)
+  * Clean-up the settings hierarchy for issue_indexer queue (#16001)
+  * Change default queue settings to be low go-routines (#15964)
+  * Improve assets handler middleware (#15961)
+  * Rename StaticUrlPrefix to AssetUrlPrefix (#15779)
+  * Use a generic markup class to display externally rendered files and diffs (#15735)
+  * Add frontend testing, require node 12 (#15315)
+  * Add `/assets` as root dir of public files (#15219)
```
Comment (noerw) [inline CHANGELOG.md]: This is misleading.
```suggestion
  * Move (custom) assets into subpath `/assets` (#15219)
```
Also the docs do not reflect this, as [noted on discord](https://discord.com/channels/322538954119184384/561007778139734027/864939872427311134)
(This PR in general had a bit of fallout, I'd not be suprised if people stumble over more breakage 🙃)

File: CHANGELOG.md [hunk 28]
```diff
@@ -4,6 +4,218 @@ This changelog goes through all the changes that have been made in each release
...
+  * Refactor Webhook + Add X-Hub-Signature (#16176)
+  * Add asymmetric JWT signing (#16010)
+  * Clean-up the settings hierarchy for issue_indexer queue (#16001)
+  * Change default queue settings to be low go-routines (#15964)
+  * Improve assets handler middleware (#15961)
+  * Rename StaticUrlPrefix to AssetUrlPrefix (#15779)
+  * Use a generic markup class to display externally rendered files and diffs (#15735)
+  * Add frontend testing, require node 12 (#15315)
+  * Move (custom) assets into subpath `/assets` (#15219)
+  * use level config in log section when sub log section not set level (#15176)
```
Comment (lafriks) [inline CHANGELOG.md]:
```suggestion
  * Use level config in log section when sub log section not set level (#15176)
```

**Review by lafriks — APPROVED** [review]

**Comment by zeripath** [comment]
> Just a side note: What about including a quick Heads Up section what people would need to "adjust" in order to install / maintain / use the software. IMHO the "BREAKING" section is rather targeted on people who follow the development of Gitea and acts as a reminder what has been changed in this release / RC.

We put a BREAKING section in our blog posts:

https://blog.gitea.io/2021/04/gitea-1.14.0-is-released/#breaking-changes-or-potentially-breaking

I'm trying to update and make it easier to explain why a PR is breaking by changing their initial comments e.g.:

https://github.com/go-gitea/gitea/pull/16010#issue-657032242

So that when it comes to writing the blog post we can copy and paste that information into the blog.

Some of the breaking changes - and I guess you're specifically referencing #15219 here - the breaking nature only becomes apparent through testing and through people trying things.

If you find a PR is breaking - it would be helpful if you could help improve this process by explaining clearly in a comment on the PR how it is breaking so we can update and improve this.

For example - this is the first time I've heard of the robots.txt problem.

**Comment by fnetX** [comment]
@zeripath thank you for your explanation. I wasn't aware of the blog posts containing more information. I can try to assist in writing these comments if I come along them, otherwise I'll just suggest to keep in mind every information available to end users is helpful to avoid being caught by breaking things you didn't fully understand as I don't always find it obvious to get what it's about.

The blog posts + explanations in the PRs are already much more helpful than the changelog in Git, don't know why I missed this ... :v:

**Comment by 6543** [comment]
@fnetX they are managed in a different repo: https://gitea.com/gitea/blog

**Comment by zeripath** [comment]
The addition of the

 ```
## ⚠️ BREAKING ⚠️ 
```

marker in pr in the first comment is an innovation I've come up with this cycle - hopefully it's useful and others will start doing it of their own volition. 

For any pr I merge or write I try to make the first comment make some sort of sense. I've tried to suggest this to other maintainers/mergers and similarly I'm hoping this will rub off on other mergers & maintainers too.

In terms of making this CHANGELOG.md file more useful - I think it needs to remain quite succinct and as updating it can be difficult I wouldn't want to have more information in it. However I take your point that the blog post can be hard to find. Maybe we need to update the readme to suggest to reading the blog?

BTW are you aware we use a tool to generate the changelog? IIRC it's possible to use that tool to emit the first comment from the PRs too so I guess we could include the instructions on how to do that in the readme?

**Pull request merged**